kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/settings/SnapshotDownloadModal.java

```diff
@@ -98,34 +121,51 @@ public SnapshotDownloadModal() {
 
     }
 
-    public void show(Consumer<SnapshotDownloadOptions> consumer) {
-        this.snapshotDownloadConsumer = consumer;
-        this.modal.setTitle(MSGS.deviceWiregraphDownloadModalTitle());
+    /*
+     * Wiregraph Snapshot Download
+     */
+
+    public void show(Consumer<String> consumer) {
+        this.wiregraphDownloadConsumer = consumer;
+        this.downloadModal.setTitle(MSGS.deviceWiregraphDownloadModalTitle());
         this.downloadModalDescription.setText(MSGS.deviceWiregraphDownloadModalHint());
         initWiregraphDownloadButtons();
-        this.modal.show();
+        this.downloadModal.show();
     }
 
-    public void show(Consumer<SnapshotDownloadOptions> consumer, List<String> availablePids) {
-        this.snapshotDownloadConsumer = consumer;
+    private void initWiregraphDownloadButtons() {
+
+        this.downloadJson.addClickHandler(e -> {
+            this.downloadModal.hide();
+            this.wiregraphDownloadConsumer.accept("JSON");
```

Comment:
Would it make sense to externalise the JSON and XML in constants?
It should help in keeping the alignment in the code in case of changes/refactorings